### PR TITLE
fix(ojoi): Updating clients to latest versions

### PR DESCRIPTION
**Pull request by jonbjarnio**
## What

Updating the clients / domains to call external api's correctly
- Updating client configs for official-journal-api
- New client/domain official-journal-application
- xroad configuration
- adjusting UI to changes of the client

## Why

To be able to communicate with external systems

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced Official Journal of Iceland Application integration, allowing users to interact with the Official Journal of Iceland through the application.

- **Improvements**
  - Enhanced error handling in the Advert Modal to log errors to the console.
  - Updated configuration parameters for different deployment environments to support the new Official Journal of Iceland Application.

- **Bug Fixes**
  - Fixed issues related to query parameters in the Advert Modal to ensure proper search functionality.

### CODE REVIEW

**Review by obmagnusson — APPROVED** [review]
Code owned files LGTM

**Comment by jonbjarnio** [comment]
@coderabbitai review

**Comment by jonbjarnio** [comment]
@coderabbitai resolve

**Review by lodmfjord — APPROVED** [review]
LGTM

**Review by Tryggvig — APPROVED** [review]

**Comment by jonbjarnio** [comment]
@coderabbitai resolve

File: libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { GetCommentsInput } from '../models/getComments.input'
+import { GetCommentsResponse } from '../models/getComments.response'
+import { PostCommentInput } from '../models/postComment.input'
+import { PostCommentResponse } from '../models/postComment.response'
+import { PostApplicationInput } from '../models/postApplication.input'
+
+@Scopes(ApiScope.internal)
+@FeatureFlag(Features.officialJournalOfIceland)
+@Resolver()
+export class OfficialJournalOfIcelandApplicationResolver {
```
Comment (saevarma) [inline libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.resolver.ts:15]: You have added the `Scopes` decorator so you also need to add `@UseGuards(IdsUserGuard, ScopesGuard` to enable auth token validation and scope checking.

File: libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.service.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import { GetCommentsInput } from '../models/getComments.input'
+
+@Injectable()
+export class OfficialJournalOfIcelandApplicationService {
+  constructor(
+    private readonly ojoiApplicationService: OfficialJournalOfIcelandApplicationClientService,
+  ) {}
+
+  async getComments(input: GetCommentsInput) {
+    return await this.ojoiApplicationService.getComments(input)
```
Comment (saevarma) [inline libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.service.ts:14]: ~ `return await` is redundant and only required when in a try/catch
```suggestion
    return this.ojoiApplicationService.getComments(input)
```

File: libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.service.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { OfficialJournalOfIcelandApplicationClientService } from '@island.is/clients/official-journal-of-iceland/application'
+import { Injectable } from '@nestjs/common'
+import { PostCommentInput } from '../models/postComment.input'
+import { PostApplicationInput } from '../models/postApplication.input'
+import { GetCommentsInput } from '../models/getComments.input'
+
+@Injectable()
+export class OfficialJournalOfIcelandApplicationService {
```
Comment (saevarma) [inline libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.service.ts:8]: What it is the nature of the data in the methods below? Here is not user authentication used to make sure to only read and write data that the user owns?

**Review by saevarma — CHANGES_REQUESTED** [review]

File: libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.service.ts (hunk 3 above)
Comment (jonbjarnio) [reply libs/api/domains/official-journal-of-iceland-application/src/lib/ojoiApplication.service.ts:8]: Auth is yet to be implemented on the otherside, will be added in the future.

**Review by saevarma — APPROVED** [review]
🚀

**Comment by jonbjarnio** [comment]
@coderabbitai resolve

**Pull request merged**